R/error-checks.R

```diff
@@ -0,0 +1,27 @@
+#' Check if the provided object is of certain type
+#'
+#' @param object An object or a list of objects
+#' @param type String representation of the type that should be checked for
+#'
+#' @return TRUE if the object or all objects inside the list are of the given type.
+#' Only the first level of the given list is considered.
+#'
+#'
+isOfType <- function(object, type) {
+  object <- c(object)
+
+  return(all(
```

Comment:
In many places, you create a local variable and then return the variable. Wouldn't that be easier to read maybe? Or bypass the return completely. For obvious function, this is clear what is being returned